querybook/server/logic/elasticsearch.py

```diff
@@ -150,51 +159,66 @@ def _get_adhoc_query_executions_iter(batch_size=1000, session=None):
+    yield from _get_query_cell_executions_iter(
+        batch_size=batch_size, fields=fields, session=session
+    )
+    yield from _get_adhoc_query_executions_iter(
+        batch_size=batch_size, fields=fields, session=session
+    )
 
 
 @with_session
-def query_execution_to_es(query_execution, data_cell=None, session=None):
+def _get_datadoc_editors(datadoc, session=None):
+    """Return datadoc editors' uids. If there is no datadoc or datadoc is public, return empty array
+    since there is no need to compute list of editors if everyone is able to access the data"""
+    if datadoc is None or datadoc.public:
+        return []
+    editors = datadoc_logic.get_data_doc_editors_by_doc_id(
+        data_doc_id=datadoc.id, session=session
+    )
+    return [editor.uid for editor in editors]
+
+
+@with_session
+def query_execution_to_es(query_execution, data_cell=None, fields=None, session=None):
     """data_cell is added as a parameter so that bulk insert of query executions won't require
     re-retrieval of data_cell"""
-    query_execution_id = query_execution.id
-
     engine_id = query_execution.engine_id
-    engine = get_query_engine_by_id(engine_id, session=session)
-
-    table_names, _ = process_query(
-        query_execution.query, language=(engine and engine.language)
-    )
-    table_names = list(chain.from_iterable(table_names))
-
-    duration = (
-        DATETIME_TO_UTC(query_execution.completed_at)
-        - DATETIME_TO_UTC(query_execution.created_at)
-        if query_execution.completed_at is not None
-        else None
-    )
+    engine = admin_logic.get_query_engine_by_id(engine_id, session=session)
+    datadoc = data_cell.doc if data_cell else None
+
+    def get_duration():
+        return (
+            DATETIME_TO_UTC(query_execution.completed_at)
+            - DATETIME_TO_UTC(query_execution.created_at)
+            if query_execution.completed_at is not None
+            else None
+        )
 
-    environments = engine.environments
-    environment_ids = [env.id for env in environments]
-
-    title = data_cell.meta.get("title", "Untitled") if data_cell else None
-
-    expand_query_execution = {
-        "id": query_execution_id,
-        "query_type": "query_execution",
-        "title": title,
-        "environment_id": environment_ids,
-        "author_uid": query_execution.uid,
-        "engine_id": engine_id,
-        "statement_type": get_table_statement_type(query_execution.query),
-        "created_at": DATETIME_TO_UTC(query_execution.created_at),
-        "duration": duration,
-        "full_table_name": table_names,
-        "query_text": query_execution.query,
+    def get_table_names():
+        table_names, _ = lineage_lib.process_query(
+            query_execution.query, language=(engine and engine.language)
+        )
+        return list(chain.from_iterable(table_names))
+
+    field_to_generator = {
+        "id": lambda: query_execution.id,
```

Comment:
upon second thought, there is no need to make every field a function since its slightly hard to read, there is a python method to check if a variable is a function https://stackoverflow.com/questions/624926/how-do-i-detect-whether-a-python-variable-is-a-function